src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -322,6 +367,31 @@ public static ListBoxModel listCheckoutCredentials(@CheckForNull Item context, S
 
             OkHttpClient client = new OkHttpClient().setProxy(getProxy(host));
 
+            int cacheSize = GitHubSCMSource.getCacheSize();
+            Jenkins jenkins = Jenkins.getInstance();
+            if (cacheSize > 0 && jenkins != null) {
+                File cacheBase = new File(jenkins.getRootDir(),
+                        GitHubSCMProbe.class.getName() + ".cache");
+                File cacheDir = null;
+                try {
+                    MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+                    sha256.update(apiUrl.getBytes(StandardCharsets.UTF_8));
+                    sha256.update("::".getBytes(StandardCharsets.UTF_8));
+                    if (username != null) {
+                        sha256.update(username.getBytes(StandardCharsets.UTF_8));
+                    }
+                    sha256.update("::".getBytes(StandardCharsets.UTF_8));
+                    sha256.update(hash.getBytes(StandardCharsets.UTF_8));
+                    cacheDir = new File(cacheBase, Base64.encodeBase64URLSafeString(sha256.digest()));
```

Comment:
Since the `SALT` changes on every Jenkins startup, won't this leave a bunch of stale directories around?

Reply:
does it... oh dear... i thought it didn't. we'll salt with something else then

Reply:
Yep... i'll update with a revised salt